### DISCUSSION TITLE
Injecting PanacheMongoEntities?

### DISCUSSION THREAD

**Discussion by DempseyRoller** [quarkusio/quarkus#33254] [Q&A] [answered]
Hello.
I was testing injecting a PanacheMongoEntity filling it with data and persisting it. But it seems it doesn't work. It at least doesn't leave an id behind. Works fine with it's non-injected counter part. I'm sure this is something basic that I don't understand but there's zero info of wether or not a PanacheMongoEntity can even be injected at all. 
So if it's not possible: why? And if it's possible: how?

[a comment by an automated account is not shown]

**Reply by loicmathieu** [reply, marked as answer]
If by injecting you mean CDI with `@Inject` then it is not supposed to work. PanacheMongoEntity are not designed to be injected.
If you want injection, you may use repositories instead (PanacheMongoRepository).

**Reply by DempseyRoller** [reply]
That's exactly what I was looking for. I guessed that there's something in the class that prevents it from being properly `@Injected ` but no where in any tutorial or question on google, does anyone state that it indeed is so. Perhaps it's just common knowledge when it comes to CDI beans? Anyway thanks! Now I know, and can use repositories if I really need to inject an object that gets persisted.
